### DISCUSSION TITLE
1.5.0 Release

### DISCUSSION THREAD

**Discussion by vomikan** [moneymanagerex/moneymanagerex#3070] [General]
It seems time to release 1.5.0
1) Stop developing new features. All new features PR to v1.5.1 branch (It does not created yet).
2) The release version should be changed to 1.5.0-RC.1 
3)  Small code changes that do not affect the translation or typo fixes are allowed.

We can release an unstable release on github and https://sourceforge.net/projects/moneymanagerex/files/
This usually attracts the attention of a large number of users then followed bug reports.

**Comment by vomikan** [comment]
@whalley what do you think regarding public release 1.5.0-RC.1 ?

**Reply by whalley** [reply]
@vomikan I think it is ready. I'm not aware of any major issues. Let it loose and get ready to catch the bug reports.

**Reply by vomikan** [reply]
Could you please quickly check for grammatical errors and other things ?
~~https://github.com/moneymanagerex/moneymanagerex/releases/tag/untagged-193f69b3f44c52fecd84~~

https://github.com/moneymanagerex/moneymanagerex/releases/tag/untagged-e005e584ab8dd424fb4d

**Reply by whalley** [reply]
Looks fine, have moved #2918 to Features.

**Reply by whalley** [reply]
@vomikan Let me know when it is time to produce a 1.5.0-RC.1 signed build for Mac.

**Reply by vomikan** [reply]
Now

**Reply by vomikan** [reply]
I'll update this folder ASAP
https://sourceforge.net/projects/moneymanagerex/files/v1.5.0-rc.1/

**Reply by whalley** [reply]
@vomikan It is here: https://drive.google.com/uc?export=download&id=1B2chEvnFuXRvm-JOMKq8sz-rgLsA4rOu

**Comment by endrep0** [comment]
Congrats and thank you!

Question, I'm using 1.4.0-alpha.6

Can I simply upgrade to 1.5.x?

Or do I need to downgrade to 1.3.x first? ( https://github.com/moneymanagerex/moneymanagerex/issues/2353 )

**Reply by vomikan** [reply]
@endrep0 1.5.0 and 1.3.x are compatable.
You just need to downgrade to 1.3.x by the script from the case #2353 
Other words... Run 1.5.0 and use the script for downgrade.

**Reply by valsamis-d** [reply]
Awesome 1.5 update guys! Thanks 🙂 

Quick question: where does portable 1.5 RC save settings? I cannot find the mmexini.db3 file.

Thank you
Dimitrios

(*) Edit: Found it in c:\Users\<username>\AppData\Roaming\MoneyManagerEx\mmexini.db3

Shouldn't it be saved/maintained in the directory where portable mmex is copied?

**Reply by vomikan** [reply]
Here is it. I case if the file missing jast cteate an enmty one.
![image](https://user-images.githubusercontent.com/6836805/111651571-65e40600-8817-11eb-8680-92aed4db233c.png)

**Reply by valsamis-d** [reply]
@vomikan  The empty file trick worked just fine :) Thanks! 

Do you want me to create an issue for this in "Issues", just for tracking purposes?

**Reply by vomikan** [reply]
It's already there
https://github.com/moneymanagerex/moneymanagerex/issues/2590

**Comment by chinagaofeng** [comment]
Do you have any plan to support more than 2-level depth of categories? That's the only disadvantage vs jGnash, and prevent me for totally moving into MMex

**Reply by vomikan** [reply]
https://github.com/moneymanagerex/moneymanagerex/issues/1477

**Comment by vomikan** [comment]
@whalley  It seems 100% done 1.5.0
I think it's time to release it this weekend.

**Reply by whalley** [reply]
Yep, let me know when you want me to build to signed Mac version and I will send you a link. One thought, should we include the other themes you have in the set of system themes?

**Reply by vomikan** [reply]
I like this one:
[yellow-blue.mmextheme.zip](https://github.com/moneymanagerex/moneymanagerex/files/6213466/yellow-blue.mmextheme.zip)
It looks pretty enough.

Couls you include it and provide proper name, please?

**Reply by vomikan** [reply]
You seem to know a lot about cmake.
May be you can help with #2714 , #2681 and #2590

**Comment by whalley** [comment]
Have added yellow-blue, name is fine for now until we add the meta-data. 

Will add those other items  to my todo list :_)

**Comment by vomikan** [comment]
@whalley I've reverted most part of my last changes.
Please check the maximum functionality.
I hope it's all right. I would like to make a release as soon as possible.

**Comment by whalley** [comment]
Yes, checked the filtering and it is back as it was.

**Reply by whalley** [reply]
Still getting crashes (with no debug report) on "Summary of Accounts" / "Monthly" - Some times it works, others the app just bombs. Not sure if you are looking at this?

Do you think we should wait a few more days before release? Seems you have made quite a few changes over past day or so.

**Reply by vomikan** [reply]
As practice shows, bugs are discovered immediately after release.

**Reply by vomikan** [reply]
It might make sense to release RC.2.

**Reply by whalley** [reply]
Latest update seem to fix crash...

**Comment by whalley** [comment]
@vomikan Signed and notarized MacOS rc.2 build here - https://drive.google.com/uc?export=download&id=10LgYKqsfUReOwPSM3GMQ-525c7OFXd2Z

**Reply by whalley** [reply]
It would be good to now have a v1.5.0 branch that we can use for any minor fixes until the 1.5.0 release whilst we push on with other changes. What do you think?

**Reply by vomikan** [reply]
If we create 1.5.0 branch it should be present permanently after release because the v1.5.0 tag be here.
For now we can create a new branch for 1.5.1. After 1.5.0 release it may be merged into master branch.

**Reply by whalley** [reply]
Fine either way, I just noted there was a 1.3.6 branch already. We can do it either way.

**Reply by vomikan** [reply]
All right, let's go.
https://github.com/moneymanagerex/moneymanagerex/tree/v1.5.0

**Comment by vomikan** [comment]
The 1.5.0 became stable. Thanks @tactilis and @avma for testings. Seems this weekend is time to release 1.5.0 stable.
@whalley i suggest to assign for master 1.6.0-alpha 
1.5.1 would be reserved for fixes.

**Comment by whalley** [comment]
@vomikan When do you want to push out the v1.5.0 ?

**Reply by vomikan** [reply]
Yesterday? 8)
It's already released. Please send me a new build.

Releases · moneymanagerex/moneymanagerex (github.com)
<https://github.com/moneymanagerex/moneymanagerex/releases>

**Comment by whalley** [comment]
Here you go: https://drive.google.com/uc?export=download&id=19rnqYBe7x68FaBKaDWlK7r-8bLPvuGBc

**Comment by whalley** [comment]
How are we managing changes for v1.5.1? Are we using the v1.5.0 branch as a v1.5.x branch in reality? I note that we have a Tag for [v.1.5.0](https://github.com/moneymanagerex/moneymanagerex/releases/tag/v1.5.0) that is pinned to the actual v1.5.0 release. 

Perhaps we should rename the branch to v1.5.x.

**Reply by vomikan** [reply]
For me it does not metter the branch name. Agree, It has sence to rename v1.5.0 to v1.5.x.

**Comment by vomikan** [comment]
@whalley could you please fix ASAP?
https://github.com/moneymanagerex/moneymanagerex/issues/3253

**Reply by whalley** [reply]
@vomikan Think we have a few commits on master that are not on the v1.5.x branch (e.g. https://github.com/vomikan/moneymanagerex/commit/8ced3307dc71dfc0f8f79f00fbfd1989f94c795b) even though we have flagged as fixed for v.1.5.2  

We probably need to double check these or perhaps we use merge everything off v1.5.2 into master (just to make sure we have not missed anything) to create the next v1.5.2RC release.   The master branch actually looks good and I'm using it as my regular app.

What do you think?

**Reply by whalley** [reply]
@vomikan Is 1.5.2 complete now? If so, the final signed MacOS build can be found here - https://mmex.ipx.co.uk/mmex-1.5.2-Darwin-2021-05-17.dmg

**Comment by whalley** [comment]
@vomikan - I see that he v1.5.2 has no RC label. Is this correct?
